src/webgpu/api/operation/rendering/blending.spec.ts

```diff
@@ -544,6 +544,100 @@ g.test('simple_blend_constant,setting_blend_constant')
+  .desc(`Test that the blending constant is not inherited between render passes.`)
+  .fn(async t => {
+    const format = 'rgba8unorm';
+    const kSize = 1;
+    const kWhiteColorData = new Float32Array([255, 255, 255, 255]);
+    const kBlackColorData = new Float32Array([0, 0, 0, 0]);
+
+    const blendComponent = { srcFactor: 'constant', dstFactor: 'one', operation: 'add' } as const;
+    const basePipeline = t.createRenderPipelineForTest({
+      format,
+      blend: undefined,
+    });
+    const testPipeline = t.createRenderPipelineForTest({
+      format,
+      blend: { color: blendComponent, alpha: blendComponent },
+    });
+
+    const renderTarget = t.device.createTexture({
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+      size: [kSize, kSize],
+      format,
+    });
+
+    const commandEncoder = t.device.createCommandEncoder();
+    {
+      const renderPass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: renderTarget.createView(),
+            loadOp: 'load',
+            storeOp: 'store',
+          },
+        ],
+      });
+      renderPass.setPipeline(basePipeline);
+      renderPass.setBindGroup(
+        0,
+        t.createBindGroupForTest(basePipeline.getBindGroupLayout(0), kBlackColorData)
+      );
+      renderPass.draw(3);
+      renderPass.setPipeline(testPipeline);
+      renderPass.setBlendConstant({ r: 1.0, g: 1.0, b: 1.0, a: 1.0 }); // Set to white color.
+      renderPass.setBindGroup(
+        0,
+        t.createBindGroupForTest(testPipeline.getBindGroupLayout(0), kWhiteColorData)
+      );
+      renderPass.draw(3);
+      // Draw [1,1,1,1] with `src * constant + dst * 1`. The blend constant to [1,1,1,1], so the
+      // result is `[1,1,1,1] * [1,1,1,1] + [0,0,0,0] * 1` = [1,1,1,1].
+      renderPass.end();
+    }
+    {
+      const renderPass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: renderTarget.createView(),
+            loadOp: 'load',
+            storeOp: 'store',
+          },
+        ],
+      });
+      renderPass.setPipeline(basePipeline);
+      renderPass.setBindGroup(
+        0,
+        t.createBindGroupForTest(basePipeline.getBindGroupLayout(0), kBlackColorData)
+      );
+      renderPass.draw(3);
+      renderPass.setPipeline(testPipeline);
+      renderPass.setBindGroup(
+        0,
+        t.createBindGroupForTest(testPipeline.getBindGroupLayout(0), kWhiteColorData)
+      );
+      renderPass.draw(3);
+      // Draw [1,1,1,1] with `src * constant + dst * 1`. The blend constant defaults to [0,0,0,0],
+      // so the result is `[1,1,1,1] * [0,0,0,0] + [0,0,0,0] * 1` = [0,0,0,0].
+      renderPass.end();
+    }
+    t.device.queue.submit([commandEncoder.finish()]);
+
+    // Check that the blend constant is not inherited from the first render pass.
+    const expColor = { R: 0, G: 0, B: 0, A: 0 };
```

Comment:
I understand this test is to ensure that the expected color should be black drawn by the second render pass without being affected by the first render pass drawing. But, a failure happens if `basePipeline` is not set and drawn first. Any thoughts on why this test should set `basePipeline` and draw it?